### PR TITLE
Flyway - Add an out-of-order configuration knob

### LINKED ISSUE #6790: Expose Configuration for Out-of-Order Flyway Migrations
**Description**
By default, Flyway will run all migrations in version order and throw an error if a new migration is introduced with a number lower than the last applied migration.  There is a boolean configuration option to allow Flyway to run migrations out of order.

**Implementation ideas**
Would it be possible to expose this as a configuration property in `application.properties` similar to how, for example, `quarkus.flyway.migrate-at-start` works.

### PR DESCRIPTION
**Pull request by machi1990**
Fixes https://github.com/quarkusio/quarkus/issues/6790

### CODE REVIEW

File: extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java [hunk 1]
```diff
@@ -165,6 +165,18 @@ void testValidateOnMigrate() {
         assertEquals(runtimeConfig.validateOnMigrate, true);
     }
 
+    @Test
+    @DisplayName("outOfOrder is correctly set")
+    void testOutOfOrder() {
+        runtimeConfig.outOfOrder = false;
+        creator = new FlywayCreator(runtimeConfig, buildConfig);
+        assertEquals(false, createdFlywayConfig().isOutOfOrder());
```
Comment (gastaldi) [inline extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java]: Not a big deal but you can use `assertFalse` here

File: extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java [hunk 2]
```diff
@@ -165,6 +165,18 @@ void testValidateOnMigrate() {
...
+
+        runtimeConfig.outOfOrder = true;
+        creator = new FlywayCreator(runtimeConfig, buildConfig);
+        assertEquals(true, createdFlywayConfig().isOutOfOrder());
```
Comment (gsmet) [inline extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java]: And `assertTrue` here.

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks good. We let you fix the assertions?

File: extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java (hunk 1 above)
Comment (machi1990) [reply extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java]: Yeap, let's do it!

File: extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java (hunk 2 above)
Comment (machi1990) [reply extensions/flyway/runtime/src/test/java/io/quarkus/flyway/runtime/FlywayCreatorTest.java]: Let's do it!

**Review by gsmet — APPROVED** [review]

**Pull request merged**